### PR TITLE
Swift Package Manager support

### PR DESCRIPTION
**Pull request by timdonnelly**

### CODE REVIEW

**Comment by timdonnelly** [comment]
- Adds `Package.swift` for SPM support.
- Moves snapshot tests to Xcode 11.2/iOS 13.2 via SPM.
- Keeps a `BlueprintUI` test pass via Cocoapods for completeness.
- Adds a bunch of missing imports that are required when building via SPM.
- Moves vector code to modern Swift `SIMD` types to avoid warnings.

File: Podfile.lock [hunk 1]
```diff
@@ -1,27 +0,0 @@
-PODS:
```
Comment (kyleve) [inline Podfile.lock]: Don't we want to keep this if the podfile is staying?

Comment (kyleve) [reply Podfile.lock]: Never mind... See there's one in the SampleApp dir.

Comment (timdonnelly) [reply Podfile.lock]: There were two, including an old unused version in `SampleApp`. `Podfile` + `.lock` now live in `SampleApp`.

**Comment by watt** [comment]
Any idea why all these snapshots changed? Seems that they fail when running in Xcode 10.x now.

**Comment by watt** [comment]
> Any idea why all these snapshots changed? Seems that they fail when running in Xcode 10.x now.

Actually it seems to be the result of running in iOS 13 instead of iOS 12.

**Comment by timdonnelly** [comment]
Yeah, I scanned them for visual differences and they seemed ok.

**Comment by watt** [comment]
My concern is that this makes `pod lib lint` fail, because it does not choose an iOS 13 simulator to run the snapshot tests.

**Comment by timdonnelly** [comment]
Which simulator does it use? It isn't specified in the `podspec`.

**Comment by timdonnelly** [comment]
Another option is to run all tests via SPM, and only perform a build step with Cocoapods.

**Pull request merged**